### PR TITLE
[aws-for-fluent-bit]  Add helm chart high performance kinesis stream plugin

### PR DESCRIPTION
**Pull request by changhyuni**
### Description of changes  
reference document: https://github.com/aws/amazon-kinesis-streams-for-fluent-bit#how-can-i-migrate-to-the-higher-performance-plugin

- Added high performance kinesis stream plugin support for aws-for-fluent-bit Helm Chart
- Updated ConfigMap and values.yaml 
- Updated README doc for Helm Chart

### Checklist
- [x] Added/modified documentation as required (such as the `README.md` for modified charts)
- [x] Incremented the chart `version` in `Chart.yaml` for the modified chart(s)
- [x] Manually tested. Describe what testing was done in the testing section below
- [x] Make sure the title of the PR is a good description that can go into the release notes

### Testing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: stable/aws-for-fluent-bit/README.md [hunk 1]
```diff
@@ -91,6 +91,15 @@ helm delete aws-for-fluent-bit --namespace kube-system
 | `firehose.timeKey` | Add the timestamp to the record under this key. By default the timestamp from Fluent Bit will not be added to records sent to Kinesis. | |
 | `firehose.timeKeyFormat` | strftime compliant format string for the timestamp; for example, `%Y-%m-%dT%H:%M:%S%z`. This option is used with `time_key`. | |
 | `firehose.extraOutputs` | Append extra outputs with value | `""` |
+| `kinesis_streams.enabled` | Whether this plugin should be enabled or not, [details](https://github.com/aws/amazon-kinesis-streams-for-fluent-bit) | `false` | ✔
```
Comment (PettitWesley) [inline stable/aws-for-fluent-bit/README.md:94]: link the C docs, not go docs: https://docs.fluentbit.io/manual/pipeline/outputs/kinesis

File: stable/aws-for-fluent-bit/README.md [hunk 2]
```diff
@@ -91,6 +91,15 @@ helm delete aws-for-fluent-bit --namespace kube-system
...
+| `kinesis_streams.region` | The AWS region. | ✔
+| `kinesis_streams.stream` | The name of the Kinesis Streams Delivery Stream that you want log records send to. | ✔
+| `kinesis_streams.sts_endpoint` | Custom endpoint for the STS API. `kinesis_streams.sts_endpoint`. | |
```
Comment (PettitWesley) [inline stable/aws-for-fluent-bit/README.md:97]: This seems to be missing a few options:
- endpoint
- role_arn

**Review by PettitWesley — CHANGES_REQUESTED** [review]
Thank you for adding this!! Few comments/fixes.

File: stable/aws-for-fluent-bit/README.md (hunk 2 above)
Comment (changhyuni) [reply stable/aws-for-fluent-bit/README.md:99]: Thnak you for review!
I updated readme your comment.

File: stable/aws-for-fluent-bit/README.md [hunk 3]
```diff
@@ -91,6 +91,17 @@ helm delete aws-for-fluent-bit --namespace kube-system
...
+| `kinesis_streams.endpoint` | Specify a custom endpoint for the Kinesis Streams API. | |
+| `kinesis_streams.role_arn` | ARN of an IAM role to assume (for cross account access). | |
+| `kinesis_streams.sts_endpoint` | Custom endpoint for the STS API. `kinesis_streams.sts_endpoint`. | |
+| `kinesis_streams.time_key` | Add the timestamp to the record under this key. By default the timestamp from Fluent Bit will not be added to records sent to Kinesis. | |
+| `kinesis_streams.time_key_format` |  strftime compliant format string for the timestamp; for example, the default is `%Y-%m-%dT%H:%M:%S`. Supports millisecond precision with `%3N` and supports nanosecond precision with `%9N` and `%L`; for example, adding `%3N` to support millisecond `%Y-%m-%dT%H:%M:%S.%3N`. This option is used with `time_key`. | |
+| `kinesis_streams.log_key` | By default, the whole log record will be sent to Kinesis. If you specify a key name with this option, then only the value of that key will be sent to Kinesis. For example, if you are using the Fluentd Docker log driver, you can specify `log_key` `log` and only the log message will be sent to Kinesis. | |
```
Comment (PettitWesley) [inline stable/aws-for-fluent-bit/README.md:102]: nit: Please use `log_key log` same formatting as the public docs do to make it clear

Comment (changhyuni) [reply stable/aws-for-fluent-bit/README.md]: @PettitWesley i fixed readme thank you.

File: stable/aws-for-fluent-bit/README.md [hunk 4]
```diff
@@ -91,6 +91,17 @@ helm delete aws-for-fluent-bit --namespace kube-system
 | `firehose.timeKey` | Add the timestamp to the record under this key. By default the timestamp from Fluent Bit will not be added to records sent to Kinesis. | |
 | `firehose.timeKeyFormat` | strftime compliant format string for the timestamp; for example, `%Y-%m-%dT%H:%M:%S%z`. This option is used with `time_key`. | |
 | `firehose.extraOutputs` | Append extra outputs with value | `""` |
+| `kinesis_streams.enabled` | It has all the core features of the `aws/amazon-kinesis-streams-for-fluent-bit` Golang Fluent Bit plugin released in 2019. The Golang plugin was named `kinesis`; this new high performance and highly efficient kinesis plugin is called `kinesis_streams` to prevent conflicts/confusion, [details](https://docs.fluentbit.io/manual/pipeline/outputs/kinesis) | `false` | ✔
```
Comment (PettitWesley) [inline stable/aws-for-fluent-bit/README.md:94]: `aws/amazon-kinesis-streams-for-fluent-bit` => this could be a link

**Review by PettitWesley — COMMENTED** [review]
Please make the one readme change to add a link, then please confirm for me that you have tested this, and then we should be good to go

**Comment by PettitWesley** [comment]
I may forget about this PR, @ mention me to remind me.

**Comment by changhyuni** [comment]
@PettitWesley fixed! 
What do you think about the url name "detail"?

**Review by PettitWesley — COMMENTED** [review]
@changhyuni thanks, I think the link "details" title is fine... you could also change it to "documentation" instead

**Comment by PettitWesley** [comment]
@changhyuni once we get to the final version, can you please paste the results of generating the config map from an example with values.yaml containing all kinesis_streams config options as evidence of testing. Then I can approve.

**Comment by PettitWesley** [comment]
Please @ me when ready, I have a lot on my backlog and may forget this PR otherwise

**Comment by changhyuni** [comment]
@PettitWesley I'm sorry for replying late
I tested it with my company account and it works well
I erased the company information
image

File: stable/aws-for-fluent-bit/README.md [hunk 5]
```diff
@@ -91,6 +91,17 @@ helm delete aws-for-fluent-bit --namespace kube-system
...
 | `firehose.timeKeyFormat` | strftime compliant format string for the timestamp; for example, `%Y-%m-%dT%H:%M:%S%z`. This option is used with `time_key`. | |
 | `firehose.extraOutputs` | Append extra outputs with value | `""` |
+| `kinesis_streams.enabled` | It has all the core features of the [documentation](https://github.com/aws/amazon-kinesis-streams-for-fluent-bit) Golang Fluent Bit plugin released in 2019. The Golang plugin was named `kinesis`; this new high performance and highly efficient kinesis plugin is called `kinesis_streams` to prevent conflicts/confusion, [details](https://docs.fluentbit.io/manual/pipeline/outputs/kinesis) | `false` | ✔
+| `kinesis_streams.region` | The AWS region. | ✔
+| `kinesis_streams.stream` | The name of the Kinesis Streams Delivery Stream that you want log records send to. | ✔
+| `kinesis_streams.endpoint` | Specify a custom endpoint for the Kinesis Streams API. | |
+| `kinesis_streams.role_arn` | ARN of an IAM role to assume (for cross account access). | |
+| `kinesis_streams.sts_endpoint` | Custom endpoint for the STS API. `kinesis_streams.sts_endpoint`. | |
```
[the remark this answers is not in the document]
Comment (PettitWesley) [reply stable/aws-for-fluent-bit/README.md:99]: Its redundant and not done for any other option.

**Review by PettitWesley — DISMISSED** [review]

**Comment by PettitWesley** [comment]
@changhyuni CI failed, looks like you need to bump chart version again, we did a release recently.

File: stable/aws-for-fluent-bit/README.md [hunk 6]
```diff
@@ -91,6 +91,17 @@ helm delete aws-for-fluent-bit --namespace kube-system
 | `firehose.timeKey` | Add the timestamp to the record under this key. By default the timestamp from Fluent Bit will not be added to records sent to Kinesis. | |
 | `firehose.timeKeyFormat` | strftime compliant format string for the timestamp; for example, `%Y-%m-%dT%H:%M:%S%z`. This option is used with `time_key`. | |
 | `firehose.extraOutputs` | Append extra outputs with value | `""` |
+| `kinesis_streams.enabled` | It has all the core features of the [documentation](https://github.com/aws/amazon-kinesis-streams-for-fluent-bit) Golang Fluent Bit plugin released in 2019. The Golang plugin was named `kinesis`; this new high performance and highly efficient kinesis plugin is called `kinesis_streams` to prevent conflicts/confusion, [details](https://docs.fluentbit.io/manual/pipeline/outputs/kinesis) | `false` | ✔
```
Comment (PettitWesley) [inline stable/aws-for-fluent-bit/README.md:94]: This reads a bit awkward. I'd do:

"Enable the high performance kinesis plugin; it has all the core features of the [Golang Fluent Bit](link) plugin..

**Review by PettitWesley — DISMISSED** [review]

**Comment by PettitWesley** [comment]
Screen Shot 2023-07-12 at 4 00 49 PM

@changhyuni sorry but it says there are still conflicts... 

I think you need a rebase:

```
git checkout master
git pull
git checkout update/aws-for-fluent-bit # feature branch
git rebase master
```

**Comment by changhyuni** [comment]
@PettitWesley thanks you,
I solved it

**Review by PettitWesley — APPROVED** [review]

**Comment by PettitWesley** [comment]
@changhyuni  Sorry it still says I can't merge it. Sometimes Github UI rebase does not work and you need to rebase on command line: https://github.com/aws/eks-charts/pull/951#issuecomment-1633313764

Screen Shot 2023-07-13 at 10 23 09 AM

**Pull request merged**